map/src/frame/components/GlobalFrame.js

```diff
@@ -33,18 +34,27 @@ const GlobalFrame = () => {
     const [menuInfo, setMenuInfo] = useState(null);
     const [width, height] = useWindowSize();
     const [openVisibleMenu, setOpenVisibleMenu] = useState(false);
+    const navigate = useNavigate();
+    const location = useLocation();
 
     const MAIN_MENU_SIZE = openMainMenu ? MAIN_MENU_OPEN_SIZE : MAIN_MENU_MIN_SIZE;
     const MENU_INFO_SIZE = menuInfo ? MENU_INFO_OPEN_SIZE : MENU_INFO_CLOSE_SIZE;
 
     useEffect(() => {
-        ctx.setInfoBlockWidth(MENU_INFO_SIZE);
-    });
+        if (menuInfo) {
+            ctx.setInfoBlockWidth(MENU_INFO_OPEN_SIZE);
+        }
+    }, [menuInfo]);
 
     useEffect(() => {
         if (ctx.infoBlockWidth === MENU_INFO_CLOSE_SIZE) {
             setShowInfoBlock(false);
-            setMenuInfo(null);
+            if (menuInfo !== null) {
+                setMenuInfo(null);
+            }
+            if (location.pathname !== MAIN_URL) {
+                navigate(MAIN_URL + location.hash);
```

Comment:
Possibly, the good idea is to include pageParams here to avoid "empty-params flickering"?